skybar/src/main/java/org/wtf/skybar/transform/TryCatchMethodVisitor.java

```diff
@@ -0,0 +1,195 @@
+package org.wtf.skybar.transform;
+
+import org.objectweb.asm.*;
+import org.objectweb.asm.commons.LocalVariablesSorter;
+import org.objectweb.asm.tree.AbstractInsnNode;
+import org.objectweb.asm.tree.InsnList;
+import org.objectweb.asm.tree.LineNumberNode;
+import org.wtf.skybar.registry.SkybarRegistry;
+import org.wtf.skybar.transform.util.WorkingLineNumberVisitor;
+
+import java.lang.invoke.CallSite;
+import java.lang.invoke.MethodHandles;
+import java.lang.invoke.MethodType;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Inserts instrumentation to update the SkybarRegistry on each method exit. Line visits are tracked as increments of local variables.
+ */
+class TryCatchMethodVisitor extends WorkingLineNumberVisitor {
+
+    private final String className;
+    private final int version;
+    private final int access;
+    private final String desc;
+    private final String sourceFile;
+    private final MethodVisitor mv;
+    private final InsnList instructions;
+    private final LocalVariablesSorter localVariablesSorter;
+
+    private Map<Integer, Integer> lineNumberLocals = new HashMap<>();
```

Comment:
Isn't particularly perf. sensitive, but could use an IntIntMap here since we've already got Koloboke lying around